program/tests/replenish.rs

```diff
@@ -0,0 +1,431 @@
+            &context.last_blockhash,
+            &accounts.onramp_account,
+            lamports,
+        )
+        .await;
+    }
+
+    // replenish and advance
+    replenish(&mut context, &accounts.vote_account.pubkey()).await;
+    advance_epoch(&mut context).await;
+
+    // deposit works in all cases
+    let instructions = instruction::deposit(
+        &id(),
+        &accounts.pool,
+        &accounts.alice_stake.pubkey(),
+        &accounts.alice_token,
+        &accounts.alice.pubkey(),
+        &accounts.alice.pubkey(),
+    );
+    let transaction = Transaction::new_signed_with_payer(
+        &instructions,
+        Some(&context.payer.pubkey()),
+        &[&context.payer, &accounts.alice],
+        context.last_blockhash,
+    );
+
+    context
+        .banks_client
+        .process_transaction(transaction)
+        .await
+        .unwrap();
+
+    assert!(context
+        .banks_client
+        .get_account(accounts.alice_stake.pubkey())
+        .await
+        .expect("get_account")
+        .is_none());
+
+    // onramp is active now if we transfered to it
+    let clock = context.banks_client.get_sysvar::<Clock>().await.unwrap();
+    let (_, onramp_stake, _) =
+        get_stake_account(&mut context.banks_client, &accounts.onramp_account).await;
+
+    // we require a fully active pool for any onramp state change to reduce complexity
+    // the pool for a healthy validator is never unstaked, and a fresh pool you can wait an epoch
+    // NOTE we might relax this for DepositSol, in which case this test would change
+    if fund_onramp && !reactivate_pool {
+        let stake = onramp_stake.unwrap();
+        assert_eq!(stake.delegation.activation_epoch, clock.epoch - 1);
+        assert_eq!(stake.delegation.deactivation_epoch, u64::MAX);
+    } else {
+        assert_eq!(onramp_stake, None);
+    }
+}
+
+#[derive(Clone, Copy, PartialEq, Eq, PartialOrd, Ord)]
+enum OnRampState {
+    Initialized,
+    Activating,
+    Active,
+    Deactive,
+}
+
+#[test_case(OnRampState::Initialized, true; "move_lamports_fresh")]
+#[test_case(OnRampState::Activating, false; "topup_warm")]
+#[test_case(OnRampState::Activating, true; "move_lamports_warm")]
+#[test_case(OnRampState::Active, false; "reset_hot")]
+#[test_case(OnRampState::Active, true; "move_lamports_hot")]
+#[test_case(OnRampState::Deactive, true; "move_lamports_cold")]
+#[tokio::test]
+async fn move_value_success(onramp_state: OnRampState, move_lamports: bool) {
+    let mut context = program_test(false).start_with_context().await;
+    let accounts = SinglePoolAccounts::default();
+    accounts
+        .initialize_for_deposit(&mut context, TEST_STAKE_AMOUNT, None)
+        .await;
+    advance_epoch(&mut context).await;
+
+    // active onramp can be as low as minimum_delegation but this is more convenient
+    let lamports = get_minimum_pool_balance(
+        &mut context.banks_client,
+        &context.payer,
+        &context.last_blockhash,
+    )
+    .await;
```

Comment:
Let's test the edge at minimum_delegation if it isn't too much trouble, that way we know the edge logic is correct